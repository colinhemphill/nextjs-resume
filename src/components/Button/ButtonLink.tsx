import { VariantProps } from 'class-variance-authority';
import Link, { LinkProps } from 'next/link';
import { PropsWithChildren } from 'react';
import { twMerge } from 'tailwind-merge';
import { buttonVariants } from './Button';

export type ButtonLinkProps = LinkProps &
  VariantProps<typeof buttonVariants> & { className?: string };

export default function ButtonLink({
  children,
  className,
  size,
  ...linkProps
}: PropsWithChildren<ButtonLinkProps>) {
  return (
    <Link
      className={twMerge(buttonVariants({ size }), className)}
      {...linkProps}
    >
      {children}
    </Link>
  );
}
