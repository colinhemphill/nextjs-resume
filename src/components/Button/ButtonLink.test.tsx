import { render, screen } from '@test-utils';
import { describe, expect, test } from 'vitest';
import ButtonLink from './ButtonLink';

describe('<ButtonLink />', () => {
  test('Snapshot', async () => {
    const { container } = render(
      <>
        <ButtonLink href="/">Button</ButtonLink>
        <ButtonLink href="/" size="sm">
          Button
        </ButtonLink>
        <ButtonLink href="/" size="md">
          Button
        </ButtonLink>
        <ButtonLink href="/" size="lg">
          Button
        </ButtonLink>
      </>,
    );
    expect(container).toMatchSnapshot();
  });

  test('Renders an accessible link element', async () => {
    render(<ButtonLink href="/">Home page</ButtonLink>);

    const button = screen.getByRole('link', { name: /home page/i });
    expect(button).toBeDefined();
  });
});
