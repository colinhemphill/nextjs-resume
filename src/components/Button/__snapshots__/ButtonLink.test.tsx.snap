// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ButtonLink /> > Snapshot 1`] = `
<div>
  <a
    class="flex items-center justify-center gap-2 rounded-md border-2 border-accent-7 bg-accent-1 font-medium text-accent-11 no-underline outline-none transition hover:border-accent-8 focus:border-transparent focus:bg-accent-9 focus:text-accentContrast focus:ring-4 focus:ring-accent-6 hover:focus:border-transparent text-base py-2 px-6"
    href="/"
  >
    Button
  </a>
  <a
    class="flex items-center justify-center gap-2 rounded-md border-2 border-accent-7 bg-accent-1 font-medium text-accent-11 no-underline outline-none transition hover:border-accent-8 focus:border-transparent focus:bg-accent-9 focus:text-accentContrast focus:ring-4 focus:ring-accent-6 hover:focus:border-transparent text-sm py-1 px-4"
    href="/"
  >
    Button
  </a>
  <a
    class="flex items-center justify-center gap-2 rounded-md border-2 border-accent-7 bg-accent-1 font-medium text-accent-11 no-underline outline-none transition hover:border-accent-8 focus:border-transparent focus:bg-accent-9 focus:text-accentContrast focus:ring-4 focus:ring-accent-6 hover:focus:border-transparent text-base py-2 px-6"
    href="/"
  >
    Button
  </a>
  <a
    class="flex items-center justify-center gap-2 rounded-md border-2 border-accent-7 bg-accent-1 font-medium text-accent-11 no-underline outline-none transition hover:border-accent-8 focus:border-transparent focus:bg-accent-9 focus:text-accentContrast focus:ring-4 focus:ring-accent-6 hover:focus:border-transparent text-lg py-3 px-8"
    href="/"
  >
    Button
  </a>
</div>
`;
